src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -216,8 +213,7 @@ export function DiscoverLegacy({
               {resultState === 'uninitialized' && <DiscoverUninitialized onRefresh={fetch} />}
               {/* @TODO: Solved in the Angular way to satisfy functional test - should be improved*/}
               <span style={{ display: resultState !== 'loading' ? 'none' : '' }}>
-                {fetchError && <DiscoverFetchError fetchError={fetchError} />}
```

Comment:
Is there a reason we're not showing any persistent error now? (Similar to what happens in Dashboard per panel for error states)

Reply:
`DiscoverFetchError` was specific to show only the Painless error. Other errors were only shown in a toast.
I talked to @kertal and he approved this change.

Reply:
I think it's nice to have a shared error message formatting across the applications, so treating the Painless error in a special way in Discover shouldn't be necessary. But on the other side, not showing any persistent error message in Discover could also be misleading, in case you miss the popover (having a long running query -> getting a🍸 -> coming back, the message is just "no results match your query" )

<img width="1835" alt="Bildschirmfoto 2020-09-23 um 11 28 15" src="https://user-images.githubusercontent.com/463851/93995845-cc6cf780-fd91-11ea-86c8-16ada7d33ceb.png">

I wonder if we could show a persistent error message for any error, would it be possible to show the content of the toast again if the user wants to? or show the raw error JSON?

Reply:
@lizozom @lukasolson we could solve this in a separate issue I think, since before just Painless errors were persistent, all others were displayed in a toast. But I just wonder if we could display a persistent error and e.g a full error popover button.